src/main/jbake/content/blog/2017/03/24/distributed-tracing-with-camel.adoc

```diff
@@ -0,0 +1,267 @@
+
+The second shows how an existing example (Loan Broker JMS) can be instrumented using the http://www.hawkular.org/hawkular-apm/[Hawkular APM] project.
+
+'''
+
+==== Camel OpenTracing Example
+
+This section will show how the new example, which uses a simple _logging_ tracer implementation by default, can be changed to use Zipkin.
+
+First edit each of the client/service1/service2 `pom.xml` files to:
+
+* Add properties
+
+```xml
+  <properties>
+    ...
+    <version.brave.opentracing>0.18.3</version.brave.opentracing>
+    <version.zipkin.reporter>0.6.12</version.zipkin.reporter>
+  </properties>
+```
+
+* Replace the `camel-example-opentracing-loggingtracer` dependency with:
+
+```xml
+    <dependency>
+      <groupId>io.opentracing.brave</groupId>
+      <artifactId>brave-opentracing</artifactId>
+      <version>${version.brave.opentracing}</version>
+    </dependency>
+    <dependency>
+      <groupId>io.zipkin.reporter</groupId>
+      <artifactId>zipkin-sender-urlconnection</artifactId>
+      <version>${version.zipkin.reporter}</version>
+    </dependency>
+```
+
+* Common code for ClientApplication Service1Application and ServiceApplication
+
+```java
+import brave.opentracing.BraveTracer;
+import io.opentracing.Tracer;
+import io.opentracing.contrib.global.GlobalTracer;
+import zipkin.Span;
+import zipkin.reporter.AsyncReporter;
+import zipkin.reporter.Reporter;
+import zipkin.reporter.urlconnection.URLConnectionSender;
+
+....
+
+    public static void initTracer() {
+        System.out.println("Using Zipkin Tracer");
+        String zipkinServerUrl = String.format("%s/api/v1/spans", System.getenv("ZIPKIN_SERVER_URL"));
+        Reporter<Span> reporter = AsyncReporter.builder(URLConnectionSender.create(zipkinServerUrl)).build();
+        brave.Tracer tracer = brave.Tracer.newBuilder().localServiceName("client").reporter(reporter).build();
+        GlobalTracer.register(BraveTracer.wrap(tracer));
+    }
+```
+
+* Invoking the `initTracer` method
+
+** In `ClientApplication.java` it should be called at the end of the `setupCamel` method.
+
+** In `Service1Application.java` it should be called before the `SpringApplication.run` method.
+
+** In `Service2Application.java` it should be called as the first statement in the `main` method.
+
+
+Once the changes have all been made, compile the example using `mvn compile` from the `examples/camel-example-opentracing` folder. Next step is to start the Zipkin server:
+
+```bash
+docker run -p 9411:9411 openzipkin/zipkin
+```
+
+Then follow the instructions in the REAMDE.md for running the three applications. Before running each command, you will need to run:
+
+```bash
+export ZIPKIN_SERVER_URL=http://localhost:9411
+```
+
+After the example has executed, start up the http://localhost:9411/[Zipkin UI] in a brower. Select the `client` service and you should see some traces:
```

Comment:
Main purpose of the post is to show case OpenTracing with Camel - so want to show ability to use different tracers and encourage existing Camel/Zipkin users to try out (and hopefully switch to) the camel-opentracing component. Will add some text to the summary along those lines.